pallets/gear/src/manager/journal.rs

```diff
@@ -207,9 +207,19 @@ where
 
         if self.check_program_id(&dispatch.destination()) {
             if let Some(gas_limit) = gas_limit {
-                let _ = GasHandlerOf::<T>::split_with_value(message_id, dispatch.id(), gas_limit);
+                // # Safety
+                //
+                // 1. There is no logic spliting value from the reserved nodes.
+                // 2. The `gas_limit` has been checked before.
```

Comment:
I would specify, that "before" here means inside running environment

Reply:
what about 

```
2. The `gas_limit` has been checked inside message queue processing
3. (for `split_with_value` ) The `dispatch.id() is genreated by system from new message_id`
```